apis/network/v1alpha3/types.go

```diff
@@ -218,10 +218,198 @@ type Subnet struct {
+	DestinationAddressPrefixes []string `json:"destinationAddressPrefixes,omitempty"`
+	// DestinationApplicationSecurityGroups - The application security group specified as destination.
+	DestinationApplicationSecurityGroups []ApplicationSecurityGroup `json:"destinationApplicationSecurityGroups,omitempty"`
+	// SourcePortRanges - The source port ranges.
+	SourcePortRanges []string `json:"sourcePortRanges,omitempty"`
+	// DestinationPortRanges - The destination port ranges.
+	DestinationPortRanges []string `json:"destinationPortRanges,omitempty"`
+	// Access - The network traffic is allowed or denied. Possible values include: 'SecurityRuleAccessAllow', 'SecurityRuleAccessDeny'
+	Access SecurityRuleAccess `json:"access,omitempty"`
+	// Priority - The priority of the rule. The value can be between 100 and 4096. The priority number must be unique for each rule in the collection. The lower the priority number, the higher the priority of the rule.
+	Priority int32 `json:"priority,omitempty"`
+	// Direction - The direction of the rule. The direction specifies if rule will be evaluated on incoming or outgoing traffic. Possible values include: 'SecurityRuleDirectionInbound', 'SecurityRuleDirectionOutbound'
+	Direction SecurityRuleDirection `json:"direction,omitempty"`
+	// ProvisioningState - The provisioning state of the public IP resource. Possible values are: 'Updating', 'Deleting', and 'Failed'.
+	ProvisioningState string `json:"provisioningState,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// SecurityRule network security rule.
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="STATE",type="string",JSONPath=".status.state"
+// +kubebuilder:printcolumn:name="LOCATION",type="string",JSONPath=".spec.location"
+// +kubebuilder:printcolumn:name="RECLAIM-POLICY",type="string",JSONPath=".spec.reclaimPolicy"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,azure}
+type SecurityRule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// SecurityRulePropertiesFormat - Properties of the security rule.
+	Properties SecurityRulePropertiesFormat `json:"properties,omitempty"`
+	// Name - The name of the resource that is unique within a resource group. This name can be used to access the resource.
+	Name string `json:"name,omitempty"`
+	// Etag - A unique read-only string that changes whenever the resource is updated.
+	Etag string `json:"etag,omitempty"`
+	// ID - Resource ID.
+	ID string `json:"id,omitempty"`
+}
+
+// A SecurityGroupSpec defines the desired state of a SecurityGroup.
+type SecurityGroupSpec struct {
```

Comment:
fields that are not imported from crossplane-runtime should be under `forProvider` struct (ref: https://github.com/crossplane/provider-azure/blob/b55e3a0dabc627726d2f5275d80f662650932d45/apis/cache/v1beta1/redis_types.go#L139)

Reply:
Just curious to know what is the significance of adding forProvider as it seems to be a level addition to me , and this is also not present in Virtualnetwork Resource.

Reply:
we have added forProvided in the SecurityGroupSpec Struct also.